samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/GradebookServiceHelperImpl.java

```diff
@@ -196,28 +211,59 @@ public void updateExternalAssessmentScore(AssessmentGradingData ag,
         //SAM-1562 We need to round the double score and covert to a double -DH
         double fScore = Precision.round(ag.getFinalScore(), 2);
         Double score = Double.valueOf(fScore).doubleValue();
-        points = score.toString();
-        log.info("rounded:  " + ag.getFinalScore() + " to: " + score.toString() );
+        points = getFormattedScore(score, gradebookUId);
+        log.debug("rounded:  " + ag.getFinalScore() + " to: " + score.toString() );
+    }
+
+    if (assignmentId == null) {
+      g.updateExternalAssessmentScore(gradebookUId, ag.getPublishedAssessmentId().toString(), ag.getAgentId(), points);
+    } else {
+
+        // This is the student grading it's own submission, we need to grant permissions to the student temporarily.
+        SecurityAdvisor securityAdvisor = new SecurityAdvisor() {
+            public SecurityAdvice isAllowed(String userId, String function, String reference) {
+                return "gradebook.gradeAll".equals(function) ? SecurityAdvice.ALLOWED : SecurityAdvice.PASS;
+            }
+        };
+
+        try {
+            securityService.pushAdvisor(securityAdvisor);
+            g.setAssignmentScoreString(gradebookUId, assignmentId, ag.getAgentId(), points, null);
+        } catch (Exception e) {
+            log.error("Error while grading submission {} for agent {}", assignmentId, ag.getAgentId());
+        } finally {
+            securityService.popAdvisor(securityAdvisor);
+        }
+
     }
-    g.updateExternalAssessmentScore(gradebookUId,
-      ag.getPublishedAssessmentId().toString(),
-      ag.getAgentId(),  points);
     if (testErrorHandling){
       throw new Exception("Encountered an error in update ExternalAssessmentScore.");
     }
   }
-  
+
   public void updateExternalAssessmentScores(Long publishedAssessmentId, final Map<String, Double> studentUidsToScores,
-		  org.sakaiproject.grading.api.GradingService g) throws Exception {
+    org.sakaiproject.grading.api.GradingService g) throws Exception {
```

Comment:
Spacing

Reply:
No man, it's fine

Reply:
It has the same indentation that the next line, it looks wrong tbh

Reply:
Well, I reverted this and now looks greater in Github, you okay then?